svm/src/account_loader.rs

```diff
@@ -58,17 +58,37 @@ pub(crate) enum TransactionLoadResult {
 }
 
 #[derive(PartialEq, Eq, Debug, Clone)]
-#[cfg_attr(feature = "dev-context-only-utils", derive(Default))]
+#[cfg_attr(
+    feature = "svm-internal",
+    field_qualifiers(nonce(pub), lamports_per_signature(pub),)
+)]
 pub struct CheckedTransactionDetails {
     pub(crate) nonce: Option<NonceInfo>,
     pub(crate) lamports_per_signature: u64,
+    pub(crate) compute_budget_limits: Result<ComputeBudgetLimits>,
```

Comment:
I like the approach of including this in the existing `CheckedTransactionDetails`.

Reply:
I like the idea of `ComputeBudgetLimits` being in here but I don't like that it's a `Result`. We shouldn't even go the SVM if a checked transaction didn't pass compute budget processing.

Reply:
I also didn't like that this is a  `Result`. Currently we return the error inside SVM with the error code from `process_compute_budget_instructions()`. So, we need to retain that behavior for now. The cleanup of this would require a feature gate, and probably better to do in its own PR.

Reply:
Does it actually require a feature gate?

Reply:
In error scenarios, it'll start returning error early, and possibly a different error. For example, if the transaction has multiple errors, the order in which the errors are checked and returned will be different.
It'll probably also need a similar change on FD side.

Reply:
Yeah I think that's the main reason I want you to do this change properly in another PR. We can move the error up and notify FD about the change in a separate PR which doesn't have all this other refactoring.

Reply:
That's fine. The current PR doesn't anymore have the logic to remove the dependency. Also, it doesn't change the existing behavior of transaction processing. 

I'd suggest we should do two follow on PRs after this.

1. Clean up the deps.
2. A feature gated cleanup of moving the error checking up in the bank.rs (instead of SVM).

Reply:
Works for me